libcodechecker/analyzers.py

```diff
@@ -0,0 +1,124 @@
+analyzers available in CodeChecker.
+"""
+
+import argparse
+import subprocess
+
+from libcodechecker import generic_package_context
+from libcodechecker import output_formatters
+from libcodechecker.analyze.analyzers import analyzer_types
+from libcodechecker.logger import add_verbose_arguments
+from libcodechecker.logger import LoggerFactory
+
+LOG = LoggerFactory.get_new_logger('ANALYZERS')
+
+
+def get_argparser_ctor_args():
+    """
+    This method returns a dict containing the kwargs for constructing an
+    argparse.ArgumentParser (either directly or as a subparser).
+    """
+
+    return {
+        'prog': 'CodeChecker analyzers',
+        'formatter_class': argparse.ArgumentDefaultsHelpFormatter,
+
+        # Description is shown when the command's help is queried directly
+        'description': "Get the list of available and supported analyzers, "
+                       "querying their version and actual binary executed.",
+
+        # Help is shown when the "parent" CodeChecker command lists the
+        # individual subcommands.
+        'help': "List supported and available analyzers."
+    }
+
+
+def add_arguments_to_parser(parser):
+    """
+    Add the subcommand's arguments to the given argparse.ArgumentParser.
+    """
+
+    parser.add_argument('--all',
+                        dest="all",
+                        action='store_true',
+                        default=False,
+                        required=False,
+                        help="Show all supported analyzers, not just the "
+                             "available ones.")
+
+    parser.add_argument('--details',
+                        dest="details",
+                        action='store_true',
+                        default=False,
+                        required=False,
+                        help="Show details about the analyzers, not just "
+                             "their names.")
+
+    parser.add_argument('-o', '--output',
+                        dest='output_format',
+                        required=False,
+                        default='rows',
+                        choices=output_formatters.USER_FORMATS,
+                        help="Specify the format of the output list.")
+
+    add_verbose_arguments(parser)
+    parser.set_defaults(func=main)
+
+
+def main(args):
+    """
+    List the analyzers' basic information supported by CodeChecker.
+    """
+
+    context = generic_package_context.get_context()
+    working, errored = \
+        analyzer_types.check_supported_analyzers(
+            analyzer_types.supported_analyzers,
+            context)
+
+    if args.output_format not in ['csv', 'json']:
+        if not args.details:
+            header = ['Name']
+        else:
+            header = ['Name', 'Path', 'Version']
+    else:
+        if not args.details:
+            header = ['name']
+        else:
+            header = ['name', 'path', 'version_string']
```

Comment:
Oh, you meant the other conditional. Basically it is there so output would be nice for the user, while keeping a machine-readable format in csv and json (so keys are lowercase, no space but `_`, etc.) I think it's better if we make the distinction here instead of using some obscure `util` call which automatically transforms (later on, maybe in an uninteded way?) human-readable header names to machine-readable keys.